src/extension/background-script/connectors/nwc.ts

```diff
@@ -0,0 +1,191 @@
+import { webln } from "@getalby/sdk";
+import { NostrWebLNProvider } from "@getalby/sdk/dist/webln";
+import lightningPayReq from "bolt11";
+import Hex from "crypto-js/enc-hex";
+import SHA256 from "crypto-js/sha256";
+import { Account } from "~/types";
+import Connector, {
+  CheckPaymentArgs,
+  CheckPaymentResponse,
+  ConnectPeerArgs,
+  ConnectPeerResponse,
+  ConnectorTransaction,
+  GetBalanceResponse,
+  GetInfoResponse,
+  GetTransactionsResponse,
+  KeysendArgs,
+  MakeInvoiceArgs,
+  MakeInvoiceResponse,
+  SendPaymentArgs,
+  SendPaymentResponse,
+  SignMessageArgs,
+  SignMessageResponse,
+} from "./connector.interface";
+
+interface Config {
+  nostrWalletConnectUrl: string;
+}
+
+class NWCConnector implements Connector {
+  config: Config;
+  nwc: NostrWebLNProvider;
+
+  get supportedMethods() {
+    return [
+      "getInfo",
+      "makeInvoice",
+      "sendPayment",
+      "sendPaymentAsync",
+      "getBalance",
+      "keysend",
+      "getTransactions",
+    ];
+  }
+
+  constructor(account: Account, config: Config) {
+    this.config = config;
+    this.nwc = new webln.NostrWebLNProvider({
+      nostrWalletConnectUrl: this.config.nostrWalletConnectUrl,
+    });
+  }
+
+  async init() {
+    return this.nwc.enable();
+  }
+
+  async unload() {
+    this.nwc.close();
+  }
+
+  async getInfo(): Promise<GetInfoResponse> {
+    const info = await this.nwc.getInfo();
+    return {
+      data: info.node,
+    };
+  }
+
+  async getBalance(): Promise<GetBalanceResponse> {
+    const balance = await this.nwc.getBalance();
```

Comment:
this methods might raise an error if the call fails (like for example if the method is not implemented).
so I think we need to catch them here and return a proper internal error

Reply:
we should also be able to know if the method is supported here, isn't it?

Reply:
Not here unless we call `getInfo` first somewhere in the connector, like in the `init` function. Is that fine?

Reply:
@bumi I tested it with mutiny but mutiny doesn't even support get_info yet so the TestConnection doesn't even work. I will let them know.

Reply:
EDIT: if getInfo fails it should just fallback to ["pay_invoice"] as the supported methods. Something must be wrong here, I'll check again.

Reply:
fix here: https://github.com/getAlby/js-sdk/pull/171

But it does mean that currently we have to wait for the timeout (60 seconds) before the fallback is returned. This is a bit unusable. I think maybe we should be able to provide a shorter timeout to Alby JS SDK for some NWC methods, what do you think?